ls1intum/Artemis
# `Programming exercises`: Show manual feedback first in feedback groups

### Checklist
#### General


#### Server

### Motivation and Context


Gives manual feedback a higher priority in the feedback view so it is easier to see.

### Description

Adds sorting of the feedbacks before sending them to the server according to the order of declaration in the `FeedbackType` enum.

I also renamed a function, which leads to large amount of code changes. Actual code changes running in production are `+- 7`

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Programming Exercise with manual grading enabled

1. create a programming exercise
2. create a submission and check that the automatic results are working as they did before.
3. assess the exercise through the exercise dashboard.
4. check that the manual feedback is on top of the feedback groups in the feedback detail view.

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage





### Screenshots
